src/nuclide_header.F90

```diff
@@ -993,12 +996,28 @@ subroutine nuclide_calculate_xs(this, i_sab, E, i_log_union, sqrtkT, &
                 micro_xs % reaction(j) = (ONE - f) * &
                      xs % value(i_grid - xs % threshold + 1) + &
                      f * xs % value(i_grid - xs % threshold + 2)
+              ! Check if we are below the (n,2n) and/or (n,3n) reaction thresholds to
+              ! skip remaining depletion-xs construction.
+              else
+                 if (j >= 4) then
+                       exit
+                 end if
               end if
             end associate
           end if
         end do
+        !there shouldn't be a threshold check for (n,gamma).
+        !I know this is not very clean but I don't want to overload the loop
+        !with too many conditional statements for now.
+        i_rxn = this % reaction_index(DEPLETION_RX(1))
+        if (i_rxn > 0) then
```

Comment:
Was there a reason you left the threshold check in? I would just get rid of it. Also, maybe move this to be the first thing in the `if (need_depletion_rx)` block since n,gamma is the first depletion reaction in that array.

Reply:
In another test I ran when I was working on getting around the bug related to the multiple components of certain reactions (e.g., (n,p0), (n,p1), etc for (n,p)) I found that Be7.h5 doesn't have data for the Be7's (n,gamma) reaction. I only noticed because my script crashed. I was trying to add-up all the components for all the reactions that were determined to be "summed_reactions". So for (n,disappear) and (n,absorption) I need (n,gamma) and that's why the script crashed when I got to Be7.h5. Since (n,disappear) is not as important as (n,p) or (n,alpha) I updated my script to work only on those two. Therefore, I haven't investigated further to see just how many nuclides have the same issue. I will get back to you once I have more information on this. In light of that, I left the threshold-check-in there because it seemed like the safest bet.

Reply:
hey @paulromano so I got a chance to look into it.
I found that Be7, H3 and He4 don't have (n,gamma). This isn't a big deal
because I don't recall seeing any of them as tally-nuclide-bins. Have you?
If that's the case then I will just update the branch by removing the check. I think that will take 
care of all your suggestions unless you have anything else on your mind?